application/lib/editor.dart

```diff
@@ -338,41 +338,40 @@ Widget _pad(Widget child) => Padding(
   child: child,
 );
 
-Widget _makeAtomSlot(Set<String> classes, Atom value, Atom parent, ValueSetter<Atom> onChanged, {
-  @required bool needsTree,
-  @required bool needsDifferent,
+Widget _makeAtomSlot(Set<String> classes, Atom? value, Atom parent, ValueSetter<Atom?>? onChanged, {
+  required bool needsTree,
+  required bool needsDifferent,
 }) {
-  assert(parent != null);
-  bool _ok(Atom atom) => (!needsTree || parent.canAddToTree(atom))
+  bool _ok(Atom? atom) => (!needsTree || parent.canAddToTree(atom!))
                       && (!needsDifferent || parent != atom);
   return DragTarget<Atom>(
-    onWillAccept: (Atom atom) => classes.contains(atom.className) && _ok(atom),
-    onAccept: (Atom atom) {
-      if (_ok(atom))
-        onChanged(atom);
+    onWillAcceptWithDetails: (DragTargetDetails<Atom>? atom) => classes.contains(atom!.data.className) && _ok(atom.data),
+    onAcceptWithDetails: (DragTargetDetails<Atom> atom) {
+      if (_ok(atom.data))
+        onChanged!(atom.data);
     },
-    builder: (BuildContext context, List<Atom> candidateData, List<Object> rejectedData) {
+    builder: (BuildContext context, List<Atom?> candidateData, List<Object?> rejectedData) {
       return Material(
         color: const Color(0x0A000000),
         child: Wrap(
           children: <Widget>[
             if (value != null && candidateData.isEmpty)
               _pad(AtomWidget(
                 atom: value,
-                onDelete: () { onChanged(null); },
+                onDelete: () { onChanged!(null); },
```

Comment:
if we just assume it's non-null, let's make it non-nullable

Reply:
fixed